src/group_theory/order_of_element.lean

```diff
@@ -124,21 +124,57 @@ section order_of
 
 section monoid
 variables {α} [monoid α]
+variables {H : Type u} [add_monoid H] {x : H}
 
 open_locale classical
 
+/-- `add_order_of h` is the additive order of the element `x`, i.e. the `n ≥ 1`, s. t. `n •ℕ x = 0`
+if it exists. Otherwise, i.e. if `x` is of infinite order, then `add_order_of x` is `0` by
+convention.-/
+noncomputable def add_order_of (x : H) : ℕ :=
+if h : ∃ n, 0 < n ∧ n •ℕ x = 0 then nat.find h else 0
+
 /-- `order_of a` is the order of the element `a`, i.e. the `n ≥ 1`, s.t. `a ^ n = 1` if it exists.
 Otherwise, i.e. if `a` is of infinite order, then `order_of a` is `0` by convention.-/
 noncomputable def order_of (a : α) : ℕ :=
 if h : ∃ n, 0 < n ∧ a ^ n = 1 then nat.find h else 0
 
+attribute [to_additive add_order_of] order_of
+
+@[simp] lemma add_order_of_of_mul_eq_order_of (a : α) :
+  add_order_of (additive.of_mul a) = order_of a :=
+begin
+  rw add_order_of,
+  rw order_of,
+  split_ifs;
+  refl
+end
+
+@[simp] lemma order_of_of_add_eq_add_order_of (x : H) :
+  order_of (multiplicative.of_add x) = add_order_of x :=
+begin
+  rw add_order_of,
+  rw order_of,
+  split_ifs;
+  refl,
+end
+
+lemma add_order_of_pos' {x : H} (h : ∃ n, 0 < n ∧ n •ℕ x = 0) : 0 < add_order_of x :=
+begin
+  rw add_order_of,
+  split_ifs,
+  exact (nat.find_spec h).1
+end
+
 lemma order_of_pos' {a : α} (h : ∃ n, 0 < n ∧ a ^ n = 1) : 0 < order_of a :=
 begin
   rw order_of,
   split_ifs,
   exact (nat.find_spec h).1
 end
 
+attribute [to_additive add_order_of_pos'] order_of_pos'
```

Comment:
```suggestion

@[to_additive add_order_of_pos']
lemma order_of_pos' {a : α} (h : ∃ n, 0 < n ∧ a ^ n = 1) : 0 < order_of a :=
begin
  rw order_of,
  split_ifs,
  exact (nat.find_spec h).1
end
```

and similarly below where it makes sense. Moving the attribute association to the lemma instead of a separate declarative line helps declutter the file, in my opinion.